src/main/java/io/reactivex/Completable.java

```diff
@@ -1414,6 +1414,31 @@ public final void subscribe(CompletableObserver s) {
     protected abstract void subscribeActual(CompletableObserver s);
 
     /**
+     * Subscribes a given CompletableObserver (subclass) to this Completable and returns the given
+     * CompletableObserver as is.
+     * <p>Usage example:
+     * <pre><code>
+     * Completable<Integer> source = Completable.complete().delay(1, TimeUnit.SECONDS);
+     * CompositeDisposable composite = new CompositeDisposable();
+     * 
+     * class ResourceCompletableObserver implements CompletableObserver, Disposable {
+     *     // ...
+     * }
+     * 
+     * composite.add(source.subscribeWith(new ResourceCompletableObserver()));
+     * </code></pre>
+     * @param <E> the type of the CompletableObserver to use and return
+     * @param observer the CompletableObserver (subclass) to use and return, not null
+     * @return the input {@code observer}
+     * @throws NullPointerException if {@code observer} is null
+     * @since 2.0
+     */
+    public final <E extends CompletableObserver> E subscribeWith(E observer) {
```

Comment:
will there also be an overload that only takes `onComplete`? Or something that takes 2 arguments, `onComplete` & `onError`?